kong/conf_loader.lua

```diff
@@ -357,6 +351,63 @@ local function overrides(k, default_v, file_conf, arg_conf)
   return value, k
 end
 
+-- @param value The options string to check for flags (whitespace separated)
+-- @param flags List of boolean flags to check for. 
+-- @returns 1) remainder string after all flags removed, 2) table with flag
+-- booleans, 3) sanitized flags string
+local function parse_option_flags(value, flags)
+  assert(type(value) == "string")
+  value = " " .. value .. " "
+  local sanitized = ""
+  local result = { }
+  for _, flag in ipairs(flags) do
+    local count
+    local patt = "%s" .. flag .. "%s"
+    value, count = value:gsub(patt, " ")
+    if count > 0 then
+      result[flag] = true
+      sanitized = sanitized .. " " .. flag
+    else
+      result[flag] = false
+    end
+  end
+  return pl_stringx.strip(value), result, pl_stringx.strip(sanitized)
+end
+
+-- Parses a listener address line.
+-- Supports multiple (comma separated) addresses, with 'ssl' and 'http2' flags.
+-- Pre- and postfixed whitespace as well as comma's are allowed.
+-- "off" as a first entry will return empty tables.
+-- @value list of entries (strings)
+-- @return list of parsed entries, each entry having fields `ip` (normalized string),
+-- `port` (number), `ssl` (bool), `http2` (bool), `listener` (string, full listener).
+local function parse_listeners(values)
+  local flags = { "ssl", "http2" }
+  local usage = "must be of form: [off] | <ip>:<port> [" ..
+                table.concat(flags, "] [") .. "], [... next entry ...]"
+
+  local list = {}
+  if values[1]:match("^%s*off") then
```

Comment:
This would match a value such as `offshore.com:80`, and disable proxying, going against the user's intent.